### PR TITLE
Advanced network support

### PR DESCRIPTION
**Pull request by leszko**
The main reason for this PR is to enabled using Advanced Network (and therefore SSL Mutual Authentication with health check on different non-ssl port).

Changes:
- Add `livenessProbe.path`, `livenessProbe.port`, `livenessProbe.scheme` (add the same for `readinessProbe`)
- Change `readinessProbe` endpoint to `/ready`
- Remove `hazelcast.rest` property (this is a non-backward-compatible change, however we need to do it, because 1) Advanced Network cannot work with this parameter 2) It's deprecated and actually removed [here](https://github.com/hazelcast/hazelcast/pull/15981/))
- Make REST disabled by default (non-backwards-compatible change, but makes the cluster more secure by default).

Note that:
- I have tested it works correctly with Advanced Network
- I have tested it does not drop any data while Rolling Upgrade
- It's the continuation of the work of @galibey (#79). Thanks Alexander for starting that. Good work! 👍 
- I applied all the comments from #79

### CODE REVIEW

File: stable/hazelcast-enterprise/README.md [hunk 1]
```diff
@@ -53,7 +53,6 @@ The following table lists the configurable parameters of the Hazelcast chart and
 | `cluster.memberCount`                      | Number of Hazelcast members                                                                                    | 2                                                    |
 | `hazelcast.licenseKey`                     | Hazelcast Enterprise License Key                                                                               | `nil`                                                |
 | `hazelcast.licenseKeySecretName`           | Kubernetes Secret Name, where Hazelcast Enterprise License Key is stored (can be used instead of licenseKey)   | `nil`                                                |
-| `hazelcast.rest`                           | Enable REST endpoints for Hazelcast member                                                                     | `true`                                               |
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/README.md]: What is the alternative now for users who were using REST API like `cluster.sh` or `healthcheck.sh`? Configuring REST via advanced networking in hazelcast.yaml?

File: stable/hazelcast-enterprise/templates/statefulset.yaml [hunk 2]
```diff
@@ -69,9 +69,9 @@ spec:
         {{- if .Values.livenessProbe.enabled }}
         livenessProbe:
           httpGet:
-            path: /hazelcast/health/node-state
-            port: {{ if .Values.hostPort }}{{ .Values.hostPort }}{{ else }}5701{{ end }}
-            scheme: {{ if .Values.hazelcast.ssl }}HTTPS{{ else }}HTTP{{ end }}
+            path: {{ .Values.livenessProbe.path }}
+            port: {{ if .Values.livenessProbe.port }}{{ .Values.livenessProbe.port }}{{ else if .Values.hostPort }}{{ .Values.hostPort }}{{ else }}5701{{ end }}
```
[the remark this answers is not in the document]
Comment (leszko) [reply stable/hazelcast-enterprise/templates/statefulset.yaml]: No, `.Values.linvenessProbe.port` is not defined by default. So, unless you need it, it's not defined.

File: stable/hazelcast-enterprise/templates/statefulset.yaml [hunk 3]
```diff
@@ -69,9 +69,9 @@ spec:
         {{- if .Values.livenessProbe.enabled }}
         livenessProbe:
           httpGet:
-            path: /hazelcast/health/node-state
```
[the remark this answers is not in the document]
Comment (leszko) [reply stable/hazelcast-enterprise/templates/statefulset.yaml]: right, changed.

File: stable/hazelcast-enterprise/templates/statefulset.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mesutcelik) [reply stable/hazelcast-enterprise/templates/statefulset.yaml]: ah ok. It is commented out in values.yaml. What about uncomment it with default value like `port:5701`
then you condition would be simpler.
`{{ if .Values.hostPort }}{{ .Values.hostPort }}{{ else }}{{.Values.livenessProbe.port  }} {{ end }}`

[the remark this answers is not in the document]
Comment (leszko) [reply stable/hazelcast-enterprise/templates/statefulset.yaml]: It's not the same. If you have both `hostPort` and `livenessProbe.port` defined (that will be actually the case in HZ Cloud), then we should use `livenessProbe.port` for health check.

File: stable/hazelcast-enterprise/README.md (hunk 1 above)
Comment (leszko) [reply stable/hazelcast-enterprise/README.md]: added `Notable Changes` section to README with the explanation.

**Comment by leszko** [comment]
> Hi @leszko , @galibey ,
> I added my comments. Do we have somewhere we list backward incompatibility changes like release notes. It could be needed for existing users.

Added `Notable Changes` to README.

**Pull request merged**